pkg/project/handlers.go

```diff
@@ -12,11 +12,20 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	klabels "k8s.io/apimachinery/pkg/labels"
 	"k8s.io/apimachinery/pkg/selection"
+	"k8s.io/utils/strings/slices"
 	kclient "sigs.k8s.io/controller-runtime/pkg/client"
 )
 
 func SetProjectSupportedRegions(req router.Request, resp router.Response) error {
-	req.Object.(*v1.ProjectInstance).SetDefaultRegion(apiv1.LocalRegion)
+	project := req.Object.(*v1.ProjectInstance)
+	project.SetDefaultRegion(apiv1.LocalRegion)
+	if slices.Contains(project.Status.SupportedRegions, apiv1.AllRegions) {
+		// If the project supports all regions, then ensure the default region and the local region are supported regions.
+		project.Status.SupportedRegions = []string{project.Status.DefaultRegion}
+		if project.Status.DefaultRegion != apiv1.LocalRegion {
+			project.Status.SupportedRegions = append(project.Status.SupportedRegions, apiv1.LocalRegion)
+		}
+	}
```

Comment:
Could you help me to understand this behavior? 

Looking at it from a local runtime project perspective, it would seem that setting `*` for a Project's `SupportedRegions` will result in the Project's `SupportRegions` actually being `local` and whatever `defaultRegion` is set. Is that not essentially equal to `*` since it encompasses everything?

Reply:
Talked offline and got a good explanation. Thanks @thedadams.

> We want status.supportedRegions to be a concrete list. Sort of like the "calculated all regions". If status.supportedRegions was ["*"], then a user (or a controller) would need to list all the regions to see which actual regions the project supported.